atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/knowledge/AbortedTransactionSoftCache.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.knowledge;
+
+import com.palantir.atlasdb.autobatch.Autobatchers;
+import com.palantir.atlasdb.autobatch.DisruptorAutobatcher;
+import com.palantir.atlasdb.futures.AtlasFutures;
+import com.palantir.common.streams.KeyedStream;
+import com.palantir.logsafe.Preconditions;
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+public final class AbortedTransactionSoftCache implements AutoCloseable {
+    public enum TransactionSoftCacheStatus {
+        PENDING_LOAD_FROM_RELIABLE,
+        IS_ABORTED,
+        IS_NOT_ABORTED;
+    }
+
+    private final DisruptorAutobatcher<Long, TransactionSoftCacheStatus> autobatcher;
+
+    private volatile PatchyCache patchyCache = null;
+    private final FutileTimestampStore futileTimestampStore;
+    private final KnownConcludedTransactions knownConcludedTransactions;
+
+    public AbortedTransactionSoftCache(
+            FutileTimestampStore futileTimestampStore, KnownConcludedTransactions knownConcludedTransactions) {
+        this.futileTimestampStore = futileTimestampStore;
+        this.knownConcludedTransactions = knownConcludedTransactions;
+        this.autobatcher = Autobatchers.coalescing(this::processBatch)
+                .safeLoggablePurpose("get-transaction-soft-cache-status")
+                .batchFunctionTimeout(Duration.ofSeconds(30))
+                .build();
+    }
+
+    public TransactionSoftCacheStatus getSoftCacheTransactionStatus(long startTimestamp) {
+        return AtlasFutures.getUnchecked(autobatcher.apply(startTimestamp));
```

Comment:
With what certainty are we using the values here? Right now it seems like if we're in the latest bucket we return IS_NOT_ABORTED regardless of whether the last known concluded timestamp is relevant or not, though I could be missing something

Reply:
(discussion: The broader strategy here, is that this is ALWAYS called with stuff that is known to be concluded, otherwise we will use the transactions table. In that case this will be accurate. I don't know if we want to hard fail if we see something after the last known concluded timestamp.)